staging/src/k8s.io/client-go/util/workqueue/queue.go

```diff
@@ -205,9 +205,12 @@ func (q *Type) Get() (item interface{}, shutdown bool) {
 	}
 
 	item = q.queue[0]
-	// The underlying array still exists and reference this object, so the object will not be garbage collected.
 	q.queue[0] = nil
 	q.queue = q.queue[1:]
+	
+	if len(q.queue) != 0 && cap(q.queue)/len(q.queue) > 2 {
+		q.queue = append([]t{}, q.queue...)
```

Comment:
```suggestion
		queue := make([]t, 0, len(q.eueue))
		queue = append(queue, q.queue...)
		q.queue = queue
```

I see your point. The array that backs `q.queue` only grows with many nil's in the beginning. Not sure how long it takes to make a difference in nowadays hardware, but it is great to make it better.